### PR TITLE
[WIP] Reverse engineer the sqls for the minidump tests

### PR DESCRIPTION
**Pull request by vraghavan78**

### CODE REVIEW

File: scripts/tests/CAggTest.sql [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+## ThreeStageAgg-DistinctOnDistrCol.mdp ##
+create table r (a int, b int, c int) distributed by (a);
+
+insert into r select i%2, i%5, i%5 from generate_series(1,1001232) i;
+explain select sum(distinct a) from r group by b;
+
+## ThreeStageAgg-ScalarAgg-DistinctDistrCol.mdp ##
+create table r (a int, b int, c int) distributed by (a);
+
+insert into table from select i%2, i%5, i%5 generate_series(1,1001232) i;
```
Comment (oarap) [inline scripts/tests/CAggTest.sql]: This is not a valid SQL statement. Script seems to have a little bug.

File: data/dxl/minidump/Agg-Limit.mdp [hunk 2]
```diff
@@ -1,5 +1,12 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <dxl:DXLMessage xmlns:dxl="http://greenplum.com/dxl/2010/12/">
+<!--
+create table x (i int, j int) distributed by (i);
+insert into foo select i, i%2 from generate_series(1,10) i;
```
Comment (bhuvnesh2703) [inline data/dxl/minidump/Agg-Limit.mdp]: creating table x, insert in foo?

File: data/dxl/minidump/EquivClassesUnion.mdp [hunk 3]
```diff
@@ -1,5 +1,17 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <dxl:DXLMessage xmlns:dxl="http://greenplum.com/dxl/2010/12/">
+<!--
+create table foo(a int, b int, c int) distributed by (a);
+create table bar(x int, y int, z int) distributed by (x);
+
+explain 
+(select * from foo where a = b)
+UNION
+(select * from foo where y = z);
```
Comment (bhuvnesh2703) [inline data/dxl/minidump/EquivClassesUnion.mdp]: select * from `bar` where y = z?

File: data/dxl/minidump/JOIN-int4-Eq-int2.mdp [hunk 4]
```diff
@@ -1,5 +1,16 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <dxl:DXLMessage xmlns:dxl="http://greenplum.com/dxl/2010/12/">
+<!--
+create table aint (i int);
+create table dint2 (i smallint);
+insert into aint select i from generate_series(1,100) i;
+insert into cdouble select i from generate_series(1,100) i;
```
Comment (bhuvnesh2703) [inline data/dxl/minidump/JOIN-int4-Eq-int2.mdp]: inserting in `cdouble`, table created above is `dint2`

**Comment by vraghavan78** [comment]
Closing. Will reopen if we want to get back to it.

File: data/dxl/minidump/Join_OuterChild_DistUniversal.mdp [hunk 5]
```diff
@@ -1,5 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <dxl:DXLMessage xmlns:dxl="http://greenplum.com/dxl/2010/12/">
+<!--
+create table mrs_t1(x int);
+
+insert into mrs_t1 select i from generate_series(1,10) i;
+explain select * from (SELECT 1) x where (1 < ANY (select x from mrs_t1));
```
Comment (zellerh) [inline data/dxl/minidump/Join_OuterChild_DistUniversal.mdp:7]: I don't think '>' and '>' are allowed in XML comments. For that reason, the CDATA syntax (for example, see file CTAS.mdp) is better.

**Pull request closed without merging**